### ISSUE TITLE
metasv error - Floating point exception

### ISSUE THREAD

**Issue by amaiacc** [bioinform/metasv#80]
Hi,

I'm trying to run metasv to integrate calls from breakdancer, lumpy and cnvnator.

I've installed and tested metasv v.0.4 without problems, but I get a " Floating point exception" error when trying to test it in on my data (1 sample and 1 chr).
I copy below the last lines of the log and the command I've run.

It creates the work directory and the pre_asm.vcf file, but it's empty.
I've tried combining the output of only 2 callers as well, but all the combinations give the same error.

I would appreciate if you could help me identify what is causing the error.

Thank you in advance and best regards,

amaia

    $METASV --reference $resourceDir/human_g1k_v37_decoy.fasta --outdir $WorkingDir/metasv --sample $sample --bam $WorkingDir/bam_sort/$sample.final.sorted.bam --chromosomes $chr  --cnvnator_native $WorkingDir/cnvnator/calls/$sample.calls  --breakdancer_native $WorkingDir/breakdancer/intrachr/samples61_breakdancer_o_$chr.ctx  --lumpy_vcf $WorkingDir/lumpy/$fam'_samples.vcf' --disable_assembly --filter_gaps --keep_standard_contigs

...
INFO 2015-11-12 11:46:24,844 metasv.main          SV types are set(['DEL', 'DUP', 'INV', 'INS'])
INFO 2015-11-12 11:46:24,845 metasv.main          Do merging
INFO 2015-11-12 11:46:24,845 metasv.main          Processing SVs of type DEL
INFO 2015-11-12 11:46:24,845 metasv.main          Intra-tool Merging SVs of type DEL
INFO 2015-11-12 11:46:24,845 metasv.main          First level merging for DEL for tool CNVnator
INFO 2015-11-12 11:46:24,903 metasv.main          First level merging for DEL for tool BreakDancer
INFO 2015-11-12 11:46:25,207 metasv.main          First level merging for DEL for tool Lumpy
INFO 2015-11-12 11:46:27,096 metasv.main          Inter-tool Merging SVs of type DEL
INFO 2015-11-12 11:46:27,588 metasv.main          Checking overlaps SVs of type DEL
INFO 2015-11-12 11:46:28,249 metasv.main          Processing SVs of type DUP
INFO 2015-11-12 11:46:28,249 metasv.main          Intra-tool Merging SVs of type DUP
INFO 2015-11-12 11:46:28,249 metasv.main          First level merging for DUP for tool CNVnator
INFO 2015-11-12 11:46:28,267 metasv.main          First level merging for DUP for tool Lumpy
INFO 2015-11-12 11:46:28,477 metasv.main          Inter-tool Merging SVs of type DUP
INFO 2015-11-12 11:46:28,549 metasv.main          Checking overlaps SVs of type DUP
INFO 2015-11-12 11:46:29,211 metasv.main          Processing SVs of type INV
INFO 2015-11-12 11:46:29,211 metasv.main          Intra-tool Merging SVs of type INV
INFO 2015-11-12 11:46:29,212 metasv.main          First level merging for INV for tool BreakDancer
INFO 2015-11-12 11:46:29,225 metasv.main          First level merging for INV for tool Lumpy
INFO 2015-11-12 11:46:29,229 metasv.main          Inter-tool Merging SVs of type INV
INFO 2015-11-12 11:46:29,280 metasv.main          Checking overlaps SVs of type INV
INFO 2015-11-12 11:46:29,333 metasv.main          Processing SVs of type INS
INFO 2015-11-12 11:46:29,333 metasv.main          Intra-tool Merging SVs of type INS
INFO 2015-11-12 11:46:29,333 metasv.main          First level merging for INS for tool BreakDancer
INFO 2015-11-12 11:46:36,834 metasv.main          Inter-tool Merging SVs of type INS
INFO 2015-11-12 11:46:56,726 metasv.main          Checking overlaps SVs of type INS
INFO 2015-11-12 11:47:17,721 metasv.main          Output merged VCF without assembly
Floating point exception

**Comment by marghoob** [comment]
HI Amaia,

Which version of MetaSV are you using? The floating-point error is possibly
from pysam when trying to index an empty VCF. Looks like it finds nothing
from the callers or might be a reading issues from the files. If you do not
mind, could you share the input files from CNVnator, BreakDancer and LUMPY?

Thanks for your interest in MetaSV.

Regards
Marghoob.

**Comment by amaiacc** [comment]
Hi Marghoob,
Thank you for your quick reply. I've used version v0.4.
I also got some errors and warnings from metasv.sv_interval (copied below).
I attach my files from CNVnator and LUMPY.
Thank you again for your help,
amaia


WARNING 2015-11-16 12:53:35,121 metasv.sv_interval   Skipping
Record(CHROM=1, POS=245859159, REF=N, ALT=[<DEL>]) due to small size
WARNING 2015-11-16 12:53:35,121 metasv.sv_interval   Skipping
Record(CHROM=1, POS=246657062, REF=N, ALT=[<DEL>]) due to small size
ERROR 2015-11-16 12:53:35,123 metasv.sv_interval   Ignoring record due to
missing SVTYPE or INFO field in Record(CHROM=1, POS=247291210, REF=N,
ALT=[[1:247333809[N])
ERROR 2015-11-16 12:53:35,123 metasv.sv_interval   Ignoring record due to
missing SVTYPE or INFO field in Record(CHROM=1, POS=247333809, REF=N,
ALT=[[1:247291210[N])
ERROR 2015-11-16 12:53:35,125 metasv.sv_interval   Ignoring record due to
missing SVTYPE or INFO field in Record(CHROM=1, POS=248610268, REF=N,
ALT=[[1:248808300[N])
ERROR 2015-11-16 12:53:35,125 metasv.sv_interval   Ignoring record due to
missing SVTYPE or INFO field in Record(CHROM=1, POS=248808300, REF=N,
ALT=[[1:248610268[N])
ERROR 2015-11-16 12:53:35,125 metasv.sv_interval   Ignoring record due to
missing SVTYPE or INFO field in Record(CHROM=1, POS=248604458, REF=N,
ALT=[N]1:248814248]])
ERROR 2015-11-16 12:53:35,126 metasv.sv_interval   Ignoring record due to
missing SVTYPE or INFO field in Record(CHROM=1, POS=248814248, REF=N,
ALT=[N]1:248604458]])
ERROR 2015-11-16 12:53:35,126 metasv.sv_interval   Ignoring record due to
missing SVTYPE or INFO field in Record(CHROM=1, POS=248603607, REF=N,
ALT=[N]1:248815102]])
ERROR 2015-11-16 12:53:35,126 metasv.sv_interval   Ignoring record due to
missing SVTYPE or INFO field in Record(CHROM=1, POS=248815102, REF=N,
ALT=[N]1:248603607]])
ERROR 2015-11-16 12:53:35,126 metasv.sv_interval   Ignoring record due to
missing SVTYPE or INFO field in Record(CHROM=1, POS=248596485, REF=N,
ALT=[N]1:248822178]])
ERROR 2015-11-16 12:53:35,126 metasv.sv_interval   Ignoring record due to
missing SVTYPE or INFO field in Record(CHROM=1, POS=248822178, REF=N,
ALT=[N]1:248596485]])



2015-11-13 16:44 GMT+01:00 Marghoob Mohiyuddin <notifications@github.com>:

> HI Amaia,
>
> Which version of MetaSV are you using? The floating-point error is possibly
> from pysam when trying to index an empty VCF. Looks like it finds nothing
> from the callers or might be a reading issues from the files. If you do not
> mind, could you share the input files from CNVnator, BreakDancer and LUMPY?
>
> Thanks for your interest in MetaSV.
>
> Regards
> Marghoob.
>

**Comment by marghoob** [comment]
Hi Amaia,

In order for MetaSV to work with VCF inputs, the VCF must adhere to the following specs:
* `ALT` column must be one of `<DEL>`, `<INS>`, `<DUP>`, `<DUP:TANDEM>`,` <INV>`.
* `INFO` column must have `SVTYPE` (one of `DEL`/`INS`/`DUP`/`DUP:TANDEM`/`INV`) and `SVLEN` fields. 

When these details are missing from a VCF record, it is ignored. In addition, if the SV has length smaller than 50bp, it is ignored. From the log, it looks like everything is being ignored from the input VCF. You can try to give MetaSV the raw calls from CNVnator using `--cnvnator_native` option, by the way.

**Comment by amaiacc** [comment]
Hi Marghoob,

I've checked the vcf file from my lumpy input, and it contains the ALT and SVTYPE columns specified as expected (I copy one entry). There are some SVTYPE=BND, which yielded the errors from the log, but these are the minority. Do you think it's ignoring them all because of these entries?

    1    869470    1    N    <DEL>    .    .    SVTYPE=DEL;STRANDS=+-:46;SVLEN=-743;END=870213;CIPOS=-10,74;CIEND=-56,10;CIPOS95=-3,19;CIEND95=-25,2;IMPRECISE;SU=46;PE=46;SR=0    GT:SU:PE:SR    ./.:2:2:0    ./.:16:16:0    ./.:3:3:0    ./.:9:9:0    ./.:10:10:0    ./.:3:3:0    ./.:3:3:0

I've used the --cnvnator_native option to input raw calls from CNVnator (see command below), and my input files look exactly as the ones from the meta_sv/test. So I guess there must be something else that I'm doing wrong.

I'm running:

    $METASV --reference $resourceDir/human_g1k_v37_decoy.fasta --gaps $WorkingDir/programs/metasv-0.4/metasv/resources/b37.gaps.bed --outdir $WorkingDir/metasv --sample $sample --chromosomes $chr --cnvnator_native ./$sample.calls --lumpy_vcf ./352_samples.vcf --disable_assembly --filter_gaps --keep_standard_contigs

I attached my test input files in the last message by replying to the email. Let me know how could I share them if you didn't receive them.

Thanks and best regards,

amaia

**Comment by amaiacc** [comment]
Hi Marghoob,
I've detected what was causing my error: the fasta file index I had used
contained the wrong chromosome names (i.e. as described here:
http://gatkforums.broadinstitute.org/discussion/6282/can-you-consider-updating-bundle-2-8-b37-human-g1k-v37-decoy-fasta-fai-gz
).
So that's why it generated the empty vcf.
I've specified the correct fai, and MetaSV ran without problems.
Thanks again for your help,
Best regards,
amaia

2015-11-18 0:08 GMT+01:00 Marghoob Mohiyuddin <notifications@github.com>:
